velox/dwio/common/compression/Compression.cpp

```diff
@@ -192,14 +193,160 @@ uint64_t ZlibDecompressor::decompress(
   return destLength - zstream_.avail_out;
 }
 
-class LzoDecompressor : public Decompressor {
+class LzoAndLz4DecompressorCommon : public Decompressor {
```

Comment:
Do you consider override `getDecompressedLength` to leverage compressed skipping?  It looks possible with LZ4/LZO